packages/story-utils/src/getStories.ts

```diff
@@ -0,0 +1,33 @@
+import path from 'path';
+import type { Parameters, Story } from '@storybook/react/types-6-0';
+import glob from 'glob';
+import merge from 'lodash/merge';
+
+type StoryData = {
+  componentTitle: string;
+  name: string;
+  parameters: Parameters;
+  storyFn: Story<unknown>;
+};
+
+export default function getStories(globPath = '**/*.stories.*'): StoryData[] {
+  const filePaths = glob.sync(globPath);
+
+  return filePaths.flatMap((filePath) => {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const storiesFileExports = require(path.resolve(filePath));
+    const { default: metadata, ...storiesMap } = storiesFileExports;
+
+    return Object.keys(storiesMap).map((storyName: string) => {
+      const story = storiesMap[storyName] as Story<unknown>;
```

Comment:
Since we're dynamically requiring files when this runs, there isn't any type safety here. Hence the `as` cast.

We could do some runtime validation of the stories, though 🤔 .

Reply:
Decided that this is a good idea, and that I'll do it in a follow up.